cisagov/getgov
# Issue #2667: Add waffle flags for organization requests and organization features

## Ticket

Resolves #2667 


## Changes


- Conditionally allow/deny permission depending on the waffle flags
- Conditionally remove nav bars depending on the waffle flags
- Unit tests



## Context for reviewers
This PR adds two waffle flags, `organization_requests` and `organization_members`. These flags are intended to be used to support the rollout of the upcoming organization milestones.

These flags do two things: 
1. they remove the associated nav item from the nav bar
2. they hook to the appropriate permission checks


Note: this PR is a joint effort between @rachidatecs / @dave-kennedy-ecs / @zandercymatics 


## Setup
You will need to test the following:
1. The `organization_requests` flag
2. The `organization_members` flag

Steps (in /admin):
1. Enable the `organization_feature` flag.
2. Disable the `organization_requests` flag.
3. Disable the `organization_members` flag.
4. Create a portfolio record.
5. Create a new UserPortfolioPermission record with yourself and the portfolio you just created. **Give yourself the admin role**.

Steps (home):
4. Ensure that you only see "Domains" and your portfolio name in the nav bar.

Steps (/admin)
5. Enable the `organization_requests` flag.
6. Enable the `organization_members` flag.

Steps (home)
7. Ensure that you can see both the `Domain requests` and `Members` nav options. Note that members currently doesn't redirect to anything and this is expected.


## Code Review Verification Steps

### As the original developer, I have

#### Satisfied acceptance criteria and met development standards


#### Ensured code standards are met (Original Developer)


#### Validated user-facing changes (if applicable)


### As a code reviewer, I have

#### Reviewed, tested, and left feedback about the changes


#### Ensured code standards are met (Code reviewer)


#### Validated user-facing changes as a developer




**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

### As a designer reviewer, I have

#### Verified that the changes match the design intention


#### Validated user-facing changes as a designer




## Screenshots
